packages/amplify-cli/src/index.ts

```diff
@@ -153,18 +153,35 @@ export const run = async (startTime: number): Promise<void> => {
   await displayBannerMessages(input);
   await executeCommand(context);
 
-  const exitCode = process.exitCode || 0;
-  if (exitCode === 0) {
-    context.usageData.emitSuccess();
-  }
-
   // no command supplied defaults to help, give update notification at end of execution
   if (input.command === 'help') {
     // Checks for available update, defaults to a 1 day interval for notification
     notify({ defer: true, isGlobal: true });
   }
 
-  await saveAllEnvParams();
+  if (context.input.command === 'push') {
+    const { providers } = stateManager.getProjectConfig(undefined, { throwIfNotExist: false, default: {} });
+    const CloudFormationProviderName = 'awscloudformation';
+    let uploaderHandler = undefined;
+    if (Array.isArray(providers) && providers.find((value) => value === CloudFormationProviderName)) {
+      uploaderHandler = await context.amplify.invokePluginMethod(
```

Comment:
since `amplify-cli` already depends directly on `amplify-provider-awscloudformation` could we just invoke this function directly?

Reply:
It turns out that this breaks the CLI when importing the provider package directly.

The problem is that this [code](https://github.com/aws-amplify/amplify-cli/blob/dev/packages/amplify-provider-awscloudformation/src/aws-predictions-regions.js#L2) invokes methods on the FeatureFlag class before it's fully initialized during import. We should fix that issue, but it's out of scope for this PR so I'm reverting this to the way it was.